src/modules/core/components/button/button.tsx

```diff
@@ -0,0 +1,65 @@
+import { type FC, useRef } from 'react';
+import clsx from 'clsx';
+// components
+import { Icon, Typography } from '@/modules/core/components';
+// hooks
+import { useRipple } from '@/modules/core/hooks/use-ripple';
+
+import type { ButtonProps, ButtonVariant } from './button.interface';
+import styles from './button.module.scss';
+
+const RIPPLE_COLORS: Record<ButtonVariant, string> = {
+  primary: styles.ripplePrimaryColor,
+  secondary: styles.rippleSecondaryColor,
+  custom: styles.rippleSecondaryColor,
+};
+
+export const Button: FC<ButtonProps> = ({
```

Comment:
I saw that you've added forward ref for Typography, maybe then we should add forwardRef here as well?

Reply:
yes, good idea. But, to make it I need `useCombinedRefs` from another PR. Suggest to make it in future when we need button ref or after we merge `time-select` PR.

Reply:
Okay